src/librustc/dep_graph/graph.rs

```diff
@@ -26,7 +26,8 @@ use super::edges::{DepGraphEdges, DepNodeIndex};
 
 #[derive(Clone)]
 pub struct DepGraph {
-    data: Option<Rc<DepGraphData>>
+    data: Option<Rc<DepGraphData>>,
+    fingerprints: Rc<RefCell<FxHashMap<DepNode, Fingerprint>>>
```

Comment:
can we use a `DepNodeIndex` and make this a vector?

Reply:
I guess not, because it is still used when incremental is disabled -- can you add some comments to that effect?

Reply:
Maybe we can at some point, but I didn't want to complicate things. Note that not all `DepNode` have an associated `Fingerprint` (e.g. anonymous nodes). Last time I checked, there where 15% anonymous nodes, which would result in an OK fill rate for a vector.

Anyway, I want to revisit this once we see how the rest of the system shakes out.

Reply:
Plus I'll add a comment for the meantime `:)`